sentinel-dashboard/src/main/webapp/resources/app/scripts/directives/header/header.js

```diff
@@ -11,8 +11,22 @@ angular.module('sentinelDashboardApp')
       restrict: 'E',
       replace: true,
       controller: function ($scope, $state, $window, AuthService) {
-        if (!$window.localStorage.getItem('session_sentinel_admin')) {
-          $state.go('login');
+        if (!$window.localStorage.getItem("session_sentinel_admin")) {
+          AuthService.check().success(function (data) {
+            if (data.code == 0) {
+              $window.localStorage.setItem('session_sentinel_admin', JSON.stringify({
+                username: data.data
+              }));
+            } else {
+              $state.go('login');
+            }
+          });
+        }
+
+        if (JSON.parse($window.localStorage.getItem("session_sentinel_admin")).username.id == 'FAKE_EMP_ID') {
```

Comment:
Hi @cdfive, I have a problem and need your guidance. 🌹
When first enter the page, the code runs is shown in numerical order
![image](https://user-images.githubusercontent.com/10150229/63504795-decc9000-c504-11e9-9bd9-ab5713d8b893.png)
So the L26 throw NPE and logout button can not hidden,
So I change code as :
```
if (!$window.localStorage.getItem("session_sentinel_admin")) {
  AuthService.check().success(function (data) {
    if (data.code == 0) {
      $window.localStorage.setItem('session_sentinel_admin', JSON.stringify({
        username: data.data
      }));
      handleLogout($scope, data.data.id)
    } else {
      $state.go('login');
    }
  });
} else {
  handleLogout($scope, JSON.parse($window.localStorage.getItem("session_sentinel_admin")).username.id)
}

function handleLogout($scope, id) {
  if (id == 'FAKE_EMP_ID') {
    $scope.showLogout = false;
  } else {
    $scope.showLogout = true;
  }
}
```
Am I writing the code correctly?

Reply:
@zhaoyuguang Yes, the code is okay!
The reason of NPE is that `AuthService.check()` is a asynchronous call, L26 execute firstly before success callback.

One more suggestion: 
We can use `JSON.stringify({data.data}))` in localStorage, just remove usename, because the data.data represent entire session data.

Reply:
Thanks for the guidance to me, the unnecessary username has been cleared, So resolve conversation